### PR TITLE
chore: enable some rules from eslint-plugin-unicorn internally

### PR DESCRIPTION
**Pull request by bmish**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: Enable some rules from eslint-plugin-unicorn internally / in eslint-config-eslint

#### What changes did you make? (Give an overview)

[eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) is a hugely popular eslint plugin (> 1M weekly downloads, > 100 contributors) with 100+ powerful rules focused especially on encouraging modern JavaScript coding practices. Disclaimer: I have contributed to this plugin and am listed as one of its maintainers.

A previous PR by @brettz9 (https://github.com/eslint/eslint/pull/15731) attempted to add most of the `unicorn/recommended` configuration from this plugin. This was rejected because the `unicorn/recommended` configuration contains many rules that could be considered too aggressive or stylistic and it was worried that frequent changes to this large configuration could become a maintenance burden.

One suggestion coming out of the previous PR was to focus on enabling a smaller set of rules, and that is what I am doing now. I have specifically chosen a small set of rules that should be minimally-controversial and clear in their benefit. In particular, the rules I have chosen are mostly focused on using more readable/efficient string/array functions, and they are all autofixable.

This enables us to get some core unicorn linting in place now, and then potentially build off of that later. If someone wants to enable additional unicorn linting, they can open separate PRs for individual consideration/discussion.

One of the rules being enabled ([unicorn/prefer-string-trim-start-end](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-trim-start-end.md)) will automate the cleanup from this earlier PR:
* https://github.com/eslint/eslint/pull/15750

### CODE REVIEW

File: tests/lib/rules/no-invalid-this.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ function extractPatterns(patterns, type) {
     }));
 
     // Flatten.
```
Comment (brettz9) [inline tests/lib/rules/no-invalid-this.js:89]:
```suggestion
```

With the semantic `flat()` now being used, I think the code is sufficient as documentation here

**Review by brettz9 — APPROVED** [review]
LGTM. Just one slight related improvement.

I think the rules such as on preferring simpler array methods make for cleaner code, and I wouldn't expect newcomers to have a hard time understanding the whys and wherefores of the rule (and they have auto-fixers).

The preferring `Set` rule comes at the cost of a little more verbosity, but IIRC, we noticed some significant performance improvements in `eslint-plugin-jsdoc` upon making such changes.

**Review by ddzz — APPROVED** [review]

**Review by aladdin-add — APPROVED** [review]

**Review by nzakas — APPROVED** [review]
Given that these are fixable, I’m ok with it.

**Review by snitin315 — APPROVED** [review]
LGTM.

**Pull request merged**